### PR TITLE
[pkg/otlp] Add basic tagger support

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

This PR adds the bare minimum support for the tagger on the OTLP ingest feature. In certain environments, if the `experimental.otlp.metrics.tag_cardinality` setting is set to `orchestrator` or `high`, metrics will be enriched with tags related to the environment (e.g. `task_arn` on ECS Fargate).

### Motivation

This is the same as the DogStatsD functionality with the `dogstatsd_tag_cardinality` setting, when the metrics have no origin set.

### Additional Notes

- Origin detection is not supported yet, but this is a small step in the right direction.
- Telemetry metrics are not enriched (this also happens on the aggregator by default).
- We have to do an ugly hack because of spf13/viper#1012 :(

### Describe how to test/QA your changes

Set the `experimental.otlp.metrics.tag_cardinality` setting to `orchestrator` and the `dogstatsd_tag_cardinality` to `orchestrator`. and check that the metrics are enriched in the same way.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by apigirl — APPROVED** [review]
👍 from docs

File: pkg/config/otlp.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ const (
 	ReceiverSubSectionKey           = "receiver"
 	ExperimentalOTLPReceiverSection = ExperimentalOTLPSection + "." + ReceiverSubSectionKey
 	ExperimentalOTLPMetrics         = ExperimentalOTLPSection + ".metrics"
+	tagCardinalityKey               = "otlp_tag_cardinality"
```
Comment (vickenty) [inline pkg/config/otlp.go:19]: Shouldn't this be `ExperimentalOTLPMetrics + ".tag_cardinality"`?

File: pkg/otlp/config.go [hunk 2]
```diff
@@ -146,6 +147,13 @@ func fromExperimentalConfig(cfg config.Config) (PipelineConfig, error) {
 		metrics = cfg.GetStringMap(config.ExperimentalOTLPMetrics)
 	}
 
+	// HACK: Because of https://github.com/spf13/viper/issues/1012
+	// we need to manually check the environment variable for any nested setting.
+	// The 'correct' solution would be to fix this in our Viper fork.
+	if env, ok := os.LookupEnv("DD_OTLP_TAG_CARDINALITY"); ok {
+		metrics["tag_cardinality"] = env
+	}
```
Comment (vickenty) [inline pkg/otlp/config.go:155]:
```suggestion
    metrics["tag_cardinality"] = cfg.GetString(config.tagCardinalityKey)
```

File: releasenotes/notes/otlp-tagger-21e4cb2b87eb5e6c.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    OTLP metrics tags are enriched when ```experimental.otlp.metrics.tag_cardinality`` is set to ``orchestrator`` or above.
```
Comment (vickenty) [inline releasenotes/notes/otlp-tagger-21e4cb2b87eb5e6c.yaml:11]: Not above, only if set to `orchestrator`, [see here](https://github.com/datadog/datadog-agent/blob/main/pkg/tagger/global.go#L264-L267)

File: pkg/config/otlp.go (hunk 1 above)
Comment (KSerrania) [reply pkg/config/otlp.go:19]: I believe this should be changed indeed, the PR description mentions `experimental.otlp.metrics.tag_cardinality`.

File: releasenotes/notes/otlp-tagger-21e4cb2b87eb5e6c.yaml [hunk 4]
```diff
@@ -8,5 +8,5 @@
 ---
 enhancements:
   - |
-    OTLP metrics tags are enriched when ```experimental.otlp.metrics.tag_cardinality`` is set to ``orchestrator`` or above.
+    OTLP metrics tags are enriched when ```experimental.otlp.metrics.tag_cardinality`` is set to ``orchestrator``.
```
Comment (vickenty) [inline releasenotes/notes/otlp-tagger-21e4cb2b87eb5e6c.yaml:11]: There's an extra backtick before experimental (three instead of two)

**Review by vickenty — APPROVED** [review]

**Review by bkabrda — APPROVED** [review]

File: pkg/config/otlp.go [hunk 5]
```diff
@@ -26,6 +27,10 @@ func SetupOTLP(config Config) {
 	config.BindEnv(ExperimentalOTLPHTTPPort, "DD_OTLP_HTTP_PORT")
 	config.BindEnv(ExperimentalOTLPgRPCPort, "DD_OTLP_GRPC_PORT")
 
+	// NOTE: This only partially works.
+	// The environment variable is also manually checked in pkg/otlp/config.go
+	config.BindEnv(ExperimentalOTLPTagCardinalityKey, "DD_OTLP_TAG_CARDINALITY")
```
Comment (albertvaka) [inline pkg/config/otlp.go:32]: `dogstatsd_tag_cardinality` defaults to `low`, should we do the same here?

Comment (mx-psi) [reply pkg/config/otlp.go:32]: This happens here
https://github.com/DataDog/datadog-agent/blob/a539e9652817a9cac232a5c80f7a98b2bac5de80/pkg/otlp/internal/serializerexporter/exporter.go#L40

Comment (albertvaka) [reply pkg/config/otlp.go:32]: Here we are reading an empty string, though:
https://github.com/DataDog/datadog-agent/blob/922d2424a2ea6c722c84d5798e0f29c366aae861/pkg/otlp/config.go#L152
From what you say I assume the later in the code the default "wins" over the empty string, but I wasn't 100% sure :)

Comment (mx-psi) [reply pkg/config/otlp.go:32]: ah, I think you are right :( and I think "" is interpreted as "whatever `dogstatsd_tag_cardinality` has, based on https://github.com/DataDog/datadog-agent/blob/a539e9652817a9cac232a5c80f7a98b2bac5de80/pkg/tagger/global.go#L282-L284 We should fix this

Comment (mx-psi) [reply pkg/config/otlp.go:32]: #10634

**Pull request merged**